Feature 82 create redux skeleton

IgorPopach opened the pull request:
We must deside if it is nessesary to create signIn async action creator...

IgorPopach committed:
created redux skeleton, authentication action

IgorPopach committed:
--amend

IgorPopach committed:
--amend

IgorPopach committed:
Merge branch 'dev' into feature_82_create_redux_skeleton

IgorPopach committed:
update starRating

IgorPopach committed:
add catch errors

IgorPopach committed:
fixed conflicts

IgorPopach committed:
fixed conflicts

IgorPopach committed:
update authreducer, add toast reducer & actions

IgorPopach committed:
update snackbar

YuliaFursovych commented:
I found a bug:
1. Go to Sign Up page and create a profile.
2. Pay attention that Profile section on header: the profile avatar should be visible.
3. Reload page with F5: profile avatar is visible.

YuliaFursovych commented on client/src/components/Rooms/RoomCard/RoomCard.js:
```diff
@@ -32,6 +33,10 @@ const roomCard = props => {
             <p>{props.description}</p>
             <Grid container justify="space-evenly" className="back-btn-wrapper">
               <span>Members: {props.members} / {props.limit}</span>
```
Should this line be deleted?

IgorPopach replied:
I will fix "Sign Up" bug in another pull request (when I will connect redux store to the SignUp component). This happens because Sign Up component only update App state (not redux store), but AppHeader watching only for redux store...
RoomCard resolved...thanks:)

YuliaFursovych commented:
One more bug: after logout add event/room button is visible, however earlier it was visible only when you authorized.

IgorPopach commented:
_One more bug: after logout add event/room button is visible, however earlier it was visible only when you authorized._ ==>
Fixed it. I added method to the SignOut button which change App state. But in future we should delete it because "add event/room" button must watching for redux store (do not take props from App component state)

IgorPopach committed:
hotFixes

IgorPopach committed:
Merge branch 'dev' into feature_82_create_redux_skeleton
